internal/backend/vars/vars.go

```diff
@@ -62,14 +80,20 @@ func (v *vars) Delete(ctx context.Context, sid sdktypes.VarScopeID, names ...sdk
 		}
 	}
 
-	return err
+	if sid.IsConnectionID() {
+		if _, err := v.conns.RefreshStatus(ctx, sid.ToConnectionID()); err != nil {
+			return err
```

Comment:
- need to add an error log here indicating we have a connection without it's vars
- or we should at least do it in one transaction
- or at least remove the connection before you remove the vars
better to have dangling vars no one can use than connections which would sure fail any usage

Reply:
- a connection might not have vars in case it doesn't need them
- we might want to remove the vars without the connection